platform/android/src/native_map_view.cpp

```diff
@@ -195,7 +200,15 @@ void NativeMapView::render() {
         mbgl::Log::Info(mbgl::Event::Android, "Not swapping as we are not ready");
     }
 
-    deactivate();
+    if (currentContext != context && currentContext != EGL_NO_CONTEXT) {
+        if (!eglMakeCurrent(currentDisplay, currentDrawSurface, currentReadSurface, currentContext)) {
+            mbgl::Log::Error(mbgl::Event::OpenGL, "eglMakeCurrent() returned error %d",
+                             eglGetError());
+            throw std::runtime_error("eglMakeCurrent() failed");
+        }
+    } else {
+        deactivate();
```

Comment:
For the record, on my Nexus 5x Android 6.0.1 `deactivate()` gets called but on my Nexus 5 Android 4.4.4 the context gets replaced with the prior context.